pennylane_lightning/lightning_tensor/quimb/_mps.py

```diff
@@ -0,0 +1,428 @@
+        "LinearCombination",
+        "Sum",
+        "SProd",
+        "Prod",
+        "Exp",
+    }
+)
+# The set of supported observables.
+
+
+def stopping_condition(op: qml.operation.Operator) -> bool:
+    """A function that determines if an operation is supported by ``lightning.tensor`` for this interface."""
+    return op.name in _operations  # pragma: no cover
+
+
+def accepted_observables(obs: qml.operation.Operator) -> bool:
+    """A function that determines if an observable is supported by ``lightning.tensor`` for this interface."""
+    return obs.name in _observables  # pragma: no cover
+
+
+def decompose_recursive(op: qml.operation.Operator) -> list:
+    """Decompose a Pennylane operator into a list of operators with at most 2 wires.
+
+    Args:
+        op (Operator): the operator to decompose.
+
+    Returns:
+        list[Operator]: a list of operators with at most 2 wires.
+    """
+
+    if len(op.wires) <= 2:
+        return [op]
+
+    decomposed_ops = []
+    for sub_op in op.decomposition():
+        decomposed_ops.extend(decompose_recursive(sub_op))
+
+    return decomposed_ops
+
+
+class QuimbMPS:
+    """Quimb MPS class.
+
+    Used internally by the `LightningTensor` device.
+    Interfaces with `quimb` for MPS manipulation, and provides methods to execute quantum circuits.
+
+    Args:
+        num_wires (int): the number of wires in the circuit.
+        interf_opts (dict): dictionary containing the interface options.
+        dtype (np.dtype): the complex type used for the MPS.
+    """
+
+    def __init__(self, num_wires, interf_opts, dtype=np.complex128):
+
+        if dtype not in [np.complex64, np.complex128]:  # pragma: no cover
+            raise TypeError(f"Unsupported complex type: {dtype}")
+
+        self._wires = Wires(range(num_wires))
+        self._dtype = dtype
+
+        self._init_state_ops = {
+            "binary": "0" * max(1, len(self._wires)),
+            "dtype": self._dtype.__name__,
+            "tags": [str(l) for l in self._wires.labels],
+        }
+
+        self._gate_opts = {
+            "contract": "swap+split",
+            "parametrize": None,
+            "cutoff": interf_opts["cutoff"],
+            "max_bond": interf_opts["max_bond_dim"],
+        }
+
+        self._expval_opts = {
+            "dtype": self._dtype.__name__,
+            "simplify_sequence": "ADCRS",
+            "simplify_atol": 0.0,
+        }
+
+        self._circuitMPS = qtn.CircuitMPS(psi0=self._initial_mps())
+
+    @property
+    def name_interf(self) -> str:
+        """The name of this interface."""
+        return "QuimbMPS interface"
+
+    @property
+    def state(self) -> qtn.MatrixProductState:
+        """Return the current MPS handled by the interface."""
+        return self._circuitMPS.psi
+
+    def state_to_array(self, digits: int = 5) -> np.ndarray:
+        """Contract the MPS into a dense array and round the values."""
+        return self._circuitMPS.to_dense().round(digits)
```

Comment:
Why are we rounding entries here? Are we expecting this to be sparse for some reason?